clang/test/Modules/no-transitive-decls-change.cppm

```diff
@@ -44,10 +44,6 @@ export inline int getA() {
     return 43;
 }
 
-export inline int getA2(int) {
```

Comment:
I am not sure why we update the tests here. Could you clarify this, maybe in the final commit description too?
(I suspect something along the lines of "we used to ignore the added declaration, but had to go with the same name and type to avoid adding new types and now it's not a problem", but just wanted to confirm).

Reply:
Previously, in `m-partA.v1.cppm`, we add a new declaration `int getA(int);`. And what we want to test is, the BMI of `useBOnly.cppm` shouldn't change after we change `m-partA.cppm` to `m-partA.v1.cppm`.

However, previously, if we lack the declaration for `int getA2(int)`, the new declaration `int getA(int)` in `m-partA.v1.cppm` would create a new function type `int *(int)`. Then (previously) the changes to type are transitive, so the BMI of `useBOnly.cppm`  will change after all. 

And now, after we made this patch, we can get rid of above logic.
So this change is not necessary but I'd like to make it since it makes the test more robust.

Reply:
Previously, in `m-partA.v1.cppm`, we add a new declaration `int getA(int);`. And what we want to test is, the BMI of `useBOnly.cppm` shouldn't change after we change `m-partA.cppm` to `m-partA.v1.cppm`.

However, previously, if we lack the declaration for `int getA2(int)`, the new declaration `int getA(int)` in `m-partA.v1.cppm` would create a new function type `int *(int)`. Then (previously) the changes to type are transitive, so the BMI of `useBOnly.cppm`  will change after all. 

And now, after we made this patch, we can get rid of above logic.
So this change is not necessary but I'd like to make it since it makes the test more robust.

Reply:
Makes sense, thanks for the explanation. +1 to making the test code simpler.